haskell/test/Lorentz/Contracts/Nettest/StablecoinClientTest.hs

```diff
@@ -35,45 +35,40 @@ test_stablecoinClientScenario =
       withEnv (`runStablecoinClient` stablecoinClientScenario)
 
 -- | Check that all the `stablecoin-client` commands work.
-
-createRole :: SpecificOrDefaultAlias -> StablecoinT m (Address, AddressOrAlias)
-createRole alias = do
-  addr <- newAddress alias
-  return (addr, AddressResolved addr)
-
 stablecoinClientScenario :: StablecoinT m ()
 stablecoinClientScenario = do
-  Moneybag nettestAddress <- asks $ ccMoneybag . SC.scCleveland
-  let originator = AddressResolved nettestAddress
-
   comment "Creating roles"
-  (masterMinterAddr, masterMinter) <- createRole "master-minter"
-  (pauserAddr, pauser) <- createRole "pauser"
-  (contractOwnerAddr, contractOwner) <- createRole "contract-owner"
-  (_, minter) <- createRole "minter"
-  (transferlistAddr, transferlist) <- createRole "transferlist"
+  originator <- refillable $ newAddress "originator"
+  masterMinter <- newAddress "master-minter"
+  pauser <- newAddress "pauser"
+  contractOwner <- newAddress "contract-owner"
+  minter <- newAddress "minter"
+  transferlist <-
+    originate "transferlist"
+      (Transferlist.Storage mempty mempty)
+      Transferlist.transferlistContract
```

Comment:
It looks like this used to be a simple address made with `createRole`:
```haskell
  (transferlistAddr, transferlist) <- createRole "transferlist"
```
which is a bit odd, any idea of why that was the case?

There is now also [a failure of this test](https://buildkite.com/serokell/stablecoin/builds/2268#0183eb5d-3c59-4403-947b-61adacbabf81), which doesn't seem related (it happens on the `deploy` following AFAICT) but perhaps it is :thinking:

Reply:
> which is a bit odd, any idea of why that was the case?

I think it may have just been an oversight on my part a long time ago :sweat_smile: 

But yea, the introduction of kinded addresses helped highlight this mistake :tada: 

> There is now also [a failure of this test](https://buildkite.com/serokell/stablecoin/builds/2268#0183eb5d-3c59-4403-947b-61adacbabf81),

Seems to be related to the change made in this comment: https://github.com/tqtezos/stablecoin/pull/198#discussion_r998097985

Not sure why `refillable` didn't work as expected, I'll have a deeper look into this.

Reply:
The failing test is fixed now